include/openspace/properties/numericalproperty.inl

```diff
@@ -395,4 +395,57 @@ std::string NumericalProperty<T>::generateAdditionalDescription() const {
     return result;
 }
 
+template <typename T>
+void NumericalProperty<T>::setInterpolationTarget(ghoul::any value) {
+    try {
+        T v = ghoul::any_cast<T>(std::move(value));
+
+        _interpolationStart = TemplateProperty<T>::_value;
+        _interpolationEnd = std::move(v);
+    }
+    catch (ghoul::bad_any_cast&) {
+        LERRORC(
+            "TemplateProperty",
+            fmt::format(
+                "Illegal cast from '{}' to '{}'",
+                value.type().name(),
+                typeid(T).name()
+            )
+        );
+    }
+}
+
+template <typename T>
+void NumericalProperty<T>::setLuaInterpolationTarget(lua_State* state) {
+    bool success = false;
+    T thisValue = PropertyDelegate<NumericalProperty<T>>::template fromLuaValue<T>(
+        state,
+        success
+    );
+    if (success) {
+        _interpolationStart = TemplateProperty<T>::_value;
+        _interpolationEnd = std::move(thisValue);
+    }
```

Comment:
Throw some type of error if !success

Reply:
No need; the "fromLuaValue" already throws an error and since the argument is the same as for the uninterpolated `setPropertyValue`, the error message still makes sense